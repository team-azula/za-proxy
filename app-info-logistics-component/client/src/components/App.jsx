import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';

import Info from './Info.jsx';
import Rating from './Rating.jsx';
import AdditionalInfo from './AdditionalInfo.jsx';

class App1 extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      app: {
      id : 1,
      name : "Concrete Shoes",
      author : "Joseph Brooks",
      imageUrl : 'https://picsum.photos/200',
      category : 'Social',
      updatedAt : '',
      size : '25 MB',
      editorChoice : true,
      rating: 3,
      ratings: 35,
      currentVersion: 1.5,
      installs : 400
  },
      id: this.props.id || 1
    }
  }

  componentDidMount() {

    let url = "/apps/";

    if (this.props.id === null || this.props.id === undefined) {
      const newId = window.location.pathname.substr(1);
      if (newId) {
        url += window.location.pathname.substr(1);
      } else {
        url += "0";
      }
    } else {
      url += this.state.id;
    }

    axios.get(url)
      .then(res => {
        this.setState({
          app: res.data
        });
      })
      .catch(err => {
        if(err) {
          console.log('Error getting data', err);
        }
      })
  }

  render() {
    return (
      <div>
        <Info app={this.state.app} />
      </div>
    )
  }
}

export default App1;
