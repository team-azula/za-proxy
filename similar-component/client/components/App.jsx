import React from "react";
import AppCards from "./AppCards.jsx";
import Header from "./Header.jsx";

class App extends React.Component {
  constructor(props) {
    super(props);
    console.log(props);
  }
  render() {
    return (
      <div className=" similar-apps container-fluid">
        <Header />
        <AppCards id={this.props.id} />
      </div>
    );
  }
}

export default App;
