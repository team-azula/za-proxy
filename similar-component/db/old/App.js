const mongoose = require('mongoose');
const db = require('./index.js');

const Schema = mongoose.Schema;

const appSchema = new Schema({
  appId: Number,
  name: String,
  logo: String,
  company: String,
  rating: Number,
  description: String
});

const App = mongoose.model('App', appSchema);

module.exports = App;