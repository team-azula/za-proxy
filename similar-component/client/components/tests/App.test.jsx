import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App.jsx';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});