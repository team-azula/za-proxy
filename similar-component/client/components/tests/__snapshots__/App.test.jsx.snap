// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className=" similar-apps container-fluid"
>
  <div
    className="header row no-gutters"
  >
    <div
      className="similar col"
    >
      <div
        className="similar"
      >
        Similar
      </div>
    </div>
    <div
      className="btn-col col"
    >
      <button
        className="seemore-btn"
      >
        See more
      </button>
    </div>
  </div>
  <div
    className="appCards row no-gutters"
  >
    <div
      className="col"
    />
  </div>
</div>
`;
